### DISCUSSION TITLE
How do you record multiple streams at once?

### DISCUSSION THREAD

**Discussion by Colorado4Wheeler** [streamlink/streamlink#3740] [Q&A] [answered]
I've got Streamlink running on my Mac and can record streams, but when I put in the command to record the stream, I see the output in console and it shows me the progress until I control-C out of it.  What if I wanted to record 5-10 streams at once?  I can open multiple terminal windows and that works fine but I'm wondering if it is possible to start the stream and have it drop back to the prompt so I can start another.  And if that is possible, how do you stop one?  Thank you!

**Comment by mkbloke** [comment]
There are at least a couple of ways to do it.  Either with shell job control and backgrounding the process (redirecting stdout/stderr from each process to either /dev/null or to file(s)), or under something like screen or tmux.  The latter is the easier way.  Take a look at: https://jeongwhanchoi.medium.com/install-tmux-on-osx-and-basics-commands-for-beginners-be22520fd95e

**Comment by mkbloke** [comment, marked as answer]
Here's an example of doing it with job control under bash for Linux.  I understand zsh is very similar to bash:
```
$ streamlink -Q -o b1.ts bbc.co.uk/iplayer/live/bbcone 396p &
[1] 2395533
$ streamlink -Q -o b2.ts bbc.co.uk/iplayer/live/bbctwo 396p &
[2] 2395562
$ jobs
[1]-  Running                 streamlink -Q -o b1.ts bbc.co.uk/iplayer/live/bbcone 396p &
[2]+  Running                 streamlink -Q -o b2.ts bbc.co.uk/iplayer/live/bbctwo 396p &
[download][b1.ts] Written 13.1 MB (1m40s @ 121.5 KB/s)                          
$ fg 1
streamlink -Q -o b1.ts bbc.co.uk/iplayer/live/bbcone 396p
^CInterrupted! Exiting...
$ jobs
[2]+  Running                 streamlink -Q -o b2.ts bbc.co.uk/iplayer/live/bbctwo 396p &
$ fg 2
streamlink -Q -o b2.ts bbc.co.uk/iplayer/live/bbctwo 396p
^CInterrupted! Exiting...
$
```
